dbms/src/Server/main.cpp

```diff
@@ -150,6 +150,12 @@ bool isClickhouseApp(const std::string & app_suffix, std::vector<char *> & argv)
 
 int main(int argc_, char ** argv_)
 {
+    if (auto r = TiFlashBuildInfo::CheckRuntimeValid(); r)
+    {
+        std::cerr << *r << std::endl;
+        exit(-1);
+    }
```

Comment:
```
$ ./tiflash version
Illegal instruction (core dumped)

$ gdb --args ./tiflash version
...
(gdb) bt
#0  0x0000000001be0193 in _GLOBAL__sub_I_CollationStringOptimized.cpp ()
#1  0x0000000001bd993d in __libc_csu_init ()
#2  0x00007ffff35424e5 in __libc_start_main () from /lib64/libc.so.6
#3  0x0000000001a95f69 in _start ()
```
TiFlash will coredump before entering `main`, I think this check doesn't help...

Reply:
After this PR, if env does not supoort required CPU instruction sets, TiFlash will cout err message and exit -1.
```
$ ./tiflash --version                                  
TiFlash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check CPU info(`/proc/cpuinfo` under Linux).

$ ./tiflash server --config xxx
TiFlash requires CPU support instruction sets: sse4.1, sse4.2, popcnt, avx, avx2, movbe. Please check CPU info(`/proc/cpuinfo` under Linux).
```

Reply:
I see. We may need to check CPU info while deploying.